src/tribler-core/run_tunnel_helper.py

```diff
@@ -90,85 +130,85 @@ async def signal_handler(sig):
-        config.chant.enabled = False
-        config.bootstrap.enabled = False
-
-        if not options.no_rest_api:
-            https = bool(options.cert_file)
-            config.api.https_enabled = https
-            config.api.http_enabled = not https
-            config.api.key = options.api_key
-
-            api_port = options.restapi
-            if "HELPER_INDEX" in os.environ and "HELPER_BASE" in os.environ:
-                api_port = int(os.environ["HELPER_BASE"]) + 10000 + int(os.environ["HELPER_INDEX"])
-            if https:
-                config.api.https_port = api_port
-                config.api.put_path_as_relative('https_certfile', options.cert_file, config.state_dir)
-            else:
-                config.api.http_port = api_port
-        else:
-            config.api.https_enabled = False
-            config.api.http_enabled = False
-
-        if options.ipv8_bootstrap_override is not None:
-            config.ipv8.bootstrap_override = options.ipv8_bootstrap_override
-
-        self.session = Session(config)
+        config = make_config(options)
+        components = list(components_gen(config))
+        session = self.session = Session(config, components)
+        session.set_as_default()
 
         self.log_circuits = options.log_circuits
-        self.session.notifier.add_observer(NTFY.TUNNEL_REMOVE, self.circuit_removed)
+        session.notifier.add_observer(NTFY.TUNNEL_REMOVE, self.circuit_removed)
 
-        await self.session.start()
+        await session.start()
 
-        if options.log_rejects:
-            # We set this after Tribler has started since the tunnel_community won't be available otherwise
-            self.session.tunnel_community.reject_callback = self.on_circuit_reject
+        with session:
+            if options.log_rejects:
+                tunnels_component = TunnelsComponent.imp()
+                tunnels_community = tunnels_component.community
+                # We set this after Tribler has started since the tunnel_community won't be available otherwise
+                tunnels_community.reject_callback = self.on_circuit_reject
 
         self.tribler_started()
-        
+
     async def stop(self):
-        await self.shutdown_task_manager()
-        if self.session:
-            return self.session.shutdown()
+        if not self._stopping:
+            self._stopping = True
+            self.session.shutdown_event.set()
+            await self.shutdown_task_manager()
+            await self.session.shutdown()
+            get_event_loop().stop()
+
+
+class PortAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        if not 0 < values < 2**16:
+            raise argparse.ArgumentError(self, "Invalid port number")
+        setattr(namespace, self.dest, values)
+
+
+class IPAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            inet_aton(values)
```

Comment:
Please consider using `ipaddress.IPv4Address()` instead of `inet_aton()`. This allows you to only catch `AddressValueError` exceptions (instead of general exceptions) and gives you a message that shows exactly what is wrong with the given IP address.

Reply:
Thanks, will made another PR with this fix